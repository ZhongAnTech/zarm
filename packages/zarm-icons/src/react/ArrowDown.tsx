import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowDown = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M494.094 733.426a41.472 41.472 0 01-20.023-11.126L140.16 388.388c-16.272-16.271-16.272-42.653 0-58.925 16.271-16.272 42.653-16.272 58.925 0L503.66 634.037 808.186 329.51c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L533.2 722.347c-10.607 10.607-25.508 14.3-39.106 11.08z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowDown);
export default ForwardRef;
