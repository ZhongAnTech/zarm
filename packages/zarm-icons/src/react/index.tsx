export { default as ArrowDown } from './ArrowDown';
export { default as ArrowLeft } from './ArrowLeft';
export { default as ArrowRight } from './ArrowRight';
export { default as ArrowUp } from './ArrowUp';
export { default as CloseCircleFill } from './CloseCircleFill';
export { default as CloseCircle } from './CloseCircle';
export { default as Close } from './Close';
export { default as DeleteKey } from './DeleteKey';
export { default as Empty } from './Empty';
export { default as Keyboard } from './Keyboard';
export { default as Minus } from './Minus';
export { default as PlusCircle } from './PlusCircle';
export { default as Plus } from './Plus';
export { default as Search } from './Search';
export { default as SuccessCircle } from './SuccessCircle';
export { default as Success } from './Success';
export { default as Volume } from './Volume';
export { default as WarningCircle } from './WarningCircle';
export { default as Warning } from './Warning';
