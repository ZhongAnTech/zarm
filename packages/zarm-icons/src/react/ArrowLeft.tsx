import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowLeft = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowLeft);
export default ForwardRef;
