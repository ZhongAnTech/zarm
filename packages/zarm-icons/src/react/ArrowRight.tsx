import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowRight = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowRight);
export default ForwardRef;
