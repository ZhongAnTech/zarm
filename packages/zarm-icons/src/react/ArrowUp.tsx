import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowUp = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M513.176 285.05a41.472 41.472 0 0120.024 11.127L867.11 630.089c16.272 16.271 16.272 42.653 0 58.925s-42.654 16.272-58.925 0L503.612 384.44 199.085 688.967c-16.272 16.271-42.654 16.271-58.925 0-16.272-16.272-16.272-42.654 0-58.926L474.07 296.129c10.606-10.606 25.508-14.299 39.105-11.078z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowUp);
export default ForwardRef;
