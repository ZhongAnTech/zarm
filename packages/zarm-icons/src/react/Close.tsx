import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgClose = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M560.149 501.223l255.344 255.344c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L501.223 560.149 245.88 815.493c-16.272 16.272-42.654 16.272-58.925 0-16.272-16.272-16.272-42.654 0-58.926l255.344-255.344L186.954 245.88c-16.272-16.272-16.272-42.654 0-58.925 16.271-16.272 42.653-16.272 58.925 0l255.344 255.344 255.344-255.344c16.272-16.272 42.654-16.272 58.926 0 16.272 16.271 16.272 42.653 0 58.925L560.149 501.223z"
          fillRule="nonzero"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgClose);
export default ForwardRef;
