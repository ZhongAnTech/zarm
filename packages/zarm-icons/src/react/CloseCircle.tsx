import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgCloseCircle = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <g fillRule="evenodd">
          <g>
            <path d="M500 460.716l137.493-137.493c10.848-10.848 28.436-10.848 39.284 0l.101.102c10.735 10.848 10.645 28.344-.203 39.08l-138.207 136.78 138.309 138.308c10.848 10.848 10.848 28.436 0 39.284-10.848 10.848-28.436 10.848-39.284 0L498.98 538.264 359.85 675.956c-10.41 10.303-27.188 10.26-37.545-.097-10.34-10.341-10.34-27.107 0-37.448L460.716 500 323.223 362.507c-10.848-10.848-10.848-28.436 0-39.284 10.848-10.848 28.436-10.848 39.284 0L500 460.716z" />
            <path
              d="M190.64 809.36c170.855 170.854 447.865 170.854 618.72 0 170.854-170.855 170.854-447.865 0-618.72-170.855-170.854-447.865-170.854-618.72 0-170.854 170.855-170.854 447.865 0 618.72zm-44.193 44.193c-195.263-195.262-195.263-511.844 0-707.106 195.262-195.263 511.844-195.263 707.106 0 195.263 195.262 195.263 511.844 0 707.106-195.262 195.263-511.844 195.263-707.106 0z"
              fillRule="nonzero"
            />
          </g>
        </g>
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgCloseCircle);
export default ForwardRef;
