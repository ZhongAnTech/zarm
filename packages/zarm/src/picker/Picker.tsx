import * as React from 'react';
import PickerView, { PickerColumnItem, PickerValue, PickerViewInstance } from '../picker-view';
import type { HTMLProps } from '../utils/utilityTypes';
import PickerContainer, { PickerContainerCssVars } from './Container';
import type { BasePickerProps } from './interface';

export interface PickerCssVars extends PickerContainerCssVars {
  '--cotnent-background': React.CSSProperties['background'];
  '--cotnent-padding': React.CSSProperties['padding'];
  '--cotnent-mask-start-background': React.CSSProperties['background'];
  '--cotnent-mask-end-background': React.CSSProperties['background'];
  '--wheel-item-rows': number;
  '--wheel-item-height': React.CSSProperties['height'];
  '--wheel-item-font-size': React.CSSProperties['fontSize'];
  '--wheel-item-text-color': React.CSSProperties['color'];
  '--wheel-item-disabled-text-color': React.CSSProperties['color'];
  '--wheel-item-selected-background': React.CSSProperties['background'];
  '--wheel-item-selected-border-radius': React.CSSProperties['borderRadius'];
}

export type PickerProps = BasePickerProps & HTMLProps<PickerCssVars>;

const Picker: React.FC<PickerProps> = (props) => {
  const {
    className,
    style,
    title,
    confirmText,
    cancelText,
    value,
    defaultValue,
    wheelDefaultValue,
    dataSource,
    fieldNames,
    itemRender,
    cols,
    maskClosable,
    mountContainer,
    forceRender,
    destroy,
    safeArea,
    onChange,
    onConfirm,
    onCancel,
    visible,
  } = props;

  const pickerViewRef = React.useRef<PickerViewInstance>(null);

  const handleChange = (changedValue: PickerValue[], items: PickerColumnItem[], index: number) => {
    visible && onChange?.(changedValue, items, index);
  };

  const handleConfirm = () => {
    onConfirm?.(pickerViewRef.current?.value!, pickerViewRef.current?.items!);
  };

  const handleCancel = () => {
    pickerViewRef.current?.reset?.();
    onCancel?.();
  };

  return (
    <PickerContainer
      title={title}
      className={className}
      style={style}
      confirmText={confirmText}
      cancelText={cancelText}
      visible={visible}
      maskClosable={maskClosable}
      forceRender={forceRender}
      destroy={destroy}
      safeArea={safeArea}
      mountContainer={mountContainer}
      onConfirm={handleConfirm}
      onCancel={handleCancel}
      onClose={handleCancel}
    >
      <PickerView
        ref={pickerViewRef}
        value={value}
        defaultValue={defaultValue}
        wheelDefaultValue={wheelDefaultValue}
        dataSource={dataSource}
        cols={cols}
        fieldNames={fieldNames}
        itemRender={itemRender}
        onChange={handleChange}
      />
    </PickerContainer>
  );
};

Picker.defaultProps = {
  dataSource: [],
  cols: Infinity,
  maskClosable: true,
  destroy: false,
};

export default Picker;
