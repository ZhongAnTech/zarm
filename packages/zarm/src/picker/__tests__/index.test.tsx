import { fireEvent,  render, screen } from '@testing-library/react';
import React from 'react';
import Button from '../../button';
import type { PickerOption } from '../../picker-view/interface';
import Picker from '../index';

describe('Picker', () => {
  it('Picker render visible', () => {
    const { container } = render(
      <Picker
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        visible
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('render custom label', () => {
    const datSource: Array<PickerOption> = [
      {
        code: '1',
        name: '北京市',
        children: [
          { code: '11', name: '海淀区' },
          { code: '12', name: '西城区' },
        ],
      },
      {
        code: '2',
        name: '上海市',
        children: [
          { code: '21', name: '黄埔区' },
          { code: '22', name: '虹口区' },
        ],
      },
    ];
    const { container } = render(
      <Picker
        dataSource={datSource}
        fieldNames={{ label: 'name', value: 'code' }}
        itemRender={(data) => data.name}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should trigger onConfirm when press ok button', () => {
    jest.useFakeTimers();
    const onConfirmFn = jest.fn();

    render(
      <Picker
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        visible
        value={['1', '12']}
        onConfirm={onConfirmFn}
      />,
    );

    const confirm = document.body.getElementsByClassName('za-picker__confirm')[0];
    fireEvent.click(confirm);
    jest.runAllTimers();
    expect(onConfirmFn).toBeCalled();
  });

  it('should trigger onCancel when press cancel button', () => {
    const onCancelFn = jest.fn();

    render(
      <Picker
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        visible
        defaultValue={['1', '12']}
        onCancel={onCancelFn}
      />,
    );

    const cancel = document.body.getElementsByClassName('za-picker__cancel')[0];
    fireEvent.click(cancel);
    expect(onCancelFn).toBeCalled();
  });

  test('imperative call onConfirm', async () => {
    const onConfirm = jest.fn();
    const onClick = async () => {
      await Picker.prompt({
        onConfirm,
        dataSource: [
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ],
      });
    };

    render(<Button onClick={onClick}>picker-prompt</Button>);
    const button = screen.getByText('picker-prompt');
    fireEvent.click(button);
    const confirm = await screen.findByText('确定');
    fireEvent.click(confirm);
    expect(onConfirm).toBeCalled();
  });

  test('imperative call onCancel', async () => {
    const onCancel = jest.fn();
    const onClick = async () => {
      await Picker.prompt({
        onCancel,
        cancelText: 'cancel',
        dataSource: [
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ],
      });
    };

    render(<Button onClick={onClick}>picker-prompt</Button>);
    const button = screen.getByText('picker-prompt');
    fireEvent.click(button);
    const cancel = await screen.findByText('cancel');
    fireEvent.click(cancel);
    expect(onCancel).toBeCalled();
  });
});
