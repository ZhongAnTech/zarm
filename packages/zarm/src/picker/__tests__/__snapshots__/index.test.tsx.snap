// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Picker render visible 1`] = `<div />`;

exports[`Picker render custom label 1`] = `<div />`;
